test/core/memory_usage/server.cc

```diff
@@ -287,19 +288,22 @@ int main(int argc, char** argv) {
                 send_status(&calls[k]);
               }
             }
-          // no break here since we want to continue to case
-          // FLING_SERVER_SEND_STATUS_SNAPSHOT to destroy the snapshot call
+            cleanup_snapshot_call = 1;
```

Comment:
Why the code refactoring instead of just having the fallthrough comment ?

Reply:
Because it's the place that needs refactoring to prevent  -Wimplicit-fallthrough warning in the entire gRPC code. Silencing the warning can also be done with `__attribute__ ((fallthrough))` (and C++14 & 17 variants of it), but those are not easily portable, so just refactoring seems less of a headache. The original construct was a bit questionable from the readability perspective anyway IMHO.

Reply:
Actually, we are using something different throughout the codebase:

https://github.com/grpc/grpc/search?utf8=%E2%9C%93&q=fallthrough&type=

Using the comment `/* fallthrough */` is the portable way to do this, and should be recognized by all compilers. We have banned the attribute usage for the reason you are describing, of course. But in locations such as https://github.com/grpc/grpc/blob/9db1691c85a720ec661266dcdcc2cbf20f7146f2/src/core/ext/transport/chttp2/transport/hpack_parser.cc#L1263 for example we do routinely use that to avoid the error.